npm/vite-dev-server/test/plugins/sourcemap.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { Plugin } from 'vite-5'
+import { ViteDevServerConfig } from '../../src/devServer'
+import { Vite } from '../../src/getVite'
+import { CypressSourcemap } from '../../src/plugins'
+import Chai, { expect } from 'chai'
+import SinonChai from 'sinon-chai'
+
+Chai.use(SinonChai)
+
+describe('sourcemap plugin', () => {
```

Comment:
Suggest also adding a test to verify non-js files don't get the source map url appended and another test to verify a js file that already contains an inline source map doesn't get updated.

Reply:
Good call: [`8607541` (#30606)](https://github.com/cypress-io/cypress/pull/30606/commits/86075410a545f84215dcaa369b0c83ce9b54a9df)